includes/multi-currency/class-frontend-prices.php

```diff
@@ -162,27 +137,30 @@ public function get_coupon_min_max_amount( $amount ) {
 	}
 
 	/**
-	 * Returns the free shipping zone settings with converted min_amount.
+	 * Returns the shipping method settings with converted amounts.
 	 *
 	 * @param array $data The shipping zone settings.
 	 *
-	 * @return array The shipping zone settings with converted min_amount.
+	 * @return array The shipping zone settings with converted amounts.
 	 */
-	public function get_free_shipping_min_amount( $data ) {
-		if ( ! isset( $data['min_amount'] ) || ! $data['min_amount'] ) {
-			return $data;
+	public function get_shipping_method_settings( $data ) {
+		if ( isset( $data['min_amount'] ) && $data['min_amount'] ) {
+			// Free shipping min amount is treated as products to avoid inconsistencies with charm pricing
+			// making a method invalid when its min amount is the same as the product's price.
+			$data['min_amount'] = $this->multi_currency->get_price( $data['min_amount'], 'product' );
+		}
+
+		if ( isset( $data['cost'] ) && $data['cost'] ) {
+			$data['cost'] = $this->multi_currency->get_price( $data['cost'], 'shipping' );
```

Comment:
Removing the review because we'll need to consider [advanced costs](https://docs.woocommerce.com/document/flat-rate-shipping/#advanced-costs) when converting them here.